README.md
SEO Basics - A TYPO3 Extension
==============================

Introduction
------------

SEO Basics is a TYPO3 CMS extension installable via TYPO3.org/TER and the extension manager.
It is compatible with TYPO3 CMS 6.2+. You can get it here: http://typo3.org/extensions/repository/view/seo_basics

What does it do?
----------------
This extension bundles most SEO features needed for getting a simple website going with search engine optimizations.
Well – so what does it do? - It adds an extra field to every page for manipulating the <title>-tag of the page.
Since this is – next to the content – the most important data for a search engine of a web page, it deserves an extra
field. If the field is left blank for a page, the regular page title will be used.
Also, the fields keywords and description are now available for the page type “standard”. Also, keywords and description
for a page as well as the last time a page was changed will be added to the output automatically.
It does not add DC tags since we believe that search engines don't care about it that much.

A new submodule under “Web” => “Info” is added to have an overview over all <title>-tags, all keywords and all
descriptions. It is possible to edit all fields at once, which is nice for comparing. When in
editing mode this page shows where there are enough (or maybe too many) keywords, and colors the
background of every field depending on the length of the content.

A new tag called the “Canonical Tag”, a tag in the head area of the web page that the search engines wanted
to avoid duplicated content.


SEO Basics also relies heavily on the realurl extension, adding a new page type that is redirected automatically when
google asks for sitemap.xml file. Right after the installation of the SEO extension there is a new page type available
that is mapped to “sitemap.xml” which means that if you enter www.your-typo3-page.com/sitemap.xml that you'll get the
google sitemap / XML sitemap for all your pages.

There is no need for an extra “google sitemap” extension, no need for an extra “metatags” extension.


Why did we create this extension?
---------------------------------
Well, first of all... we did quite some research and did some tests with pages to see how manipulating e.g. the page
title or the description affects the ranking and also the output on the search result pages. We then found out what is
important for the search engines and what they care about. Then we tried out several extensions but we noticed that they
are not quite usable e.g. several googlesitemap extensions weren't working as expected and not out of the box.
We also wanted one extension that does almost most of the things out of the box, so there's not a lot to configure.
We wanted to have an extension to get new installations up and running with SEO pretty fast, without a lot of
configuration and without dozens of not-quite-perfect extensions.


ToDo
----
* Add composer.json
* Rewrite Documentation
* Check all labels
* Release 1.0
* Add new og: fields
* Come up with a new backend module


Credits
-------
Pull Requests are very welcome!

* Thanks to EDV-Sachverständigenbüro Weißleder Stuttgart (www.weissleder.de) and ITANA (www.itana.com) for sponsoring the initial development of this extension.
* Thanks to b13 and its clients to sponsor the further development of this extension.
* Thanks to the TYPO3 community for giving feedback on the extension.
* Thanks to Jesus Christ who gave me the power and energy to live and therefore to write this extension.

ChangeLog
2014-07-02  Benjamin Mack  <benni@typo3.org>

	* [BUGFIX] Sitemap does not work with 6.2, see #58638

2014-05-07  Benjamin Mack  <benni@typo3.org>

	* [BUGFIX] Sitemap: Domain prefix also fetches hidden domains

2014-02-19  Benjamin Mack  <benni@typo3.org>

	* [BUGFIX] Sitemap should not include pages that are not on the same domain

2013-05-13  Benjamin Mack  <benni@typo3.org>

	* [RELEASE] Release of seo_basics 0.8.5
	* [CLEANUP] Set requirements to TYPO3 4.5, and added compatibility for TYPO3 6.1.x

2012-06-22  Benjamin Mack  <benni@typo3.org>

	* [FEATURE] Source Formatter: Added possibility to configure frontend output indention (Thanks to Michael Glaesser)

2011-11-11  Benjamin Mack  <benni@typo3.org>

	* Sitemap: Hide-in-Menu doesn't hide all the subpages as well anymore (Thanks to Lars Zimmermann)
	* Sitemap: Fixed the issue to disable recyclers and all their subpages (Thanks to Lars Zimmermann)
	* Sitemap: A new option allows for deactivating the resolving of shortcut root pages. This is helpful when your rootpage is a shortcut to a subpage. The option is called "resolveMainShortcut" which is set by default.: tx_seo_xmlsitemaps.10.resolveMainShortcut = 0

2011-10-17  Benjamin Mack  <benni@typo3.org>

	* Sitemap: Added functionality to crawl additional sites, and thus, allows for subpages like tt_news
	* Canonical Tag: Added functionality for mountpoints
	* Sitemap: Added XML-prologue (Thanks to Фарид Мурадимов)

2011-07-28  Benjamin Mack  <benni@typo3.org>

	* Module: Allow possibility to hide sysfolders in the overview (Thanks to Maarten Mandemaker)
	* Sitemap: Added hook for adding additional URLs (Thanks to Claus Harup)
	* Sitemap: Made Sitemap valid if XHTML cleaning is enabled (Thanks to Klaus Weidenbach)
	* Added dependency on RealURL (Thanks to Radu Dumbraveanu)
	* Sitemap: Support for multiple languages / different sitemaps (Thanks to Benjamin Bergmann)
	* Sitemap: Fixed problem with subpages of sysfolders (Thanks to Benjamin Bergmann)
	* Sitemap: Fixed small visibility for class property
	* Sitemap: Fixed double http:// in Sitemap URLs (Thanks to Oliver Rompcik)

2011-07-13  Benjamin Mack  <benni@typo3.org>

	* Fixed dependency to RealURL in the info module (Thanks to Robert Heel)

2011-07-08  Benjamin Mack  <benni@typo3.org>

	* Release of seo_basics 0.8.0
	* Improvements to the canonical tag
 	* Improvements to the sitemap and multi-domain setup
	* Fixed ext_tables.sql (Thanks to Marcel Zweifel)

2011-04-16  Benjamin Mack  <benni@typo3.org>

	* Added the canonical tag functionality.

2010-01-15  Benjamin Mack  <benni@typo3.org>

	* Release of seo_basics 0.7.0
	* Breaking change: The seo_basics metatags and sitemap TypoScript is now in a static_template and needs to be added manually through Web=>Template=>Edit the whole template record! (Thanks to Michael Cannon)
	* TYPO3 4.3: Title-Tag is now found in the tab "Metadata" on each page
	* Title-Tag: Changed titletag DB field from varchar() to tinytext (Thanks to Michael Cannon)
	* Title-Tag: Changed titletag maximum length to 70 characters (Thanks to Michael Cannon)
	* Title-Tag / Keyword module: Changed coloring of keywords (less than 3 keywords make it yellow) (Thanks to Michael Cannon)

2010-01-14  Benjamin Mack  <benni@typo3.org>

	* Source formatter: Fixed indention when having textarea elements on a page (Thanks to Sven Burkert for reporting this issue)
	* Sitemap: Fixed validation error when using tidy or source formatters (Thanks to Andreas Dörler for reporting this issue)
	* Sitemap: Fixed compatibility issue with RealURL when having domain string pointers in a configuration (Thanks to Steffen Gebert and Ingmar Schlecht)

2008-09-19  Benjamin Mack  <benni@typo3.org>

	* Release of seo_basics 0.6.0
	* Sitemap: Links are now generated by typolink (better for coolURI)
	* Sitemap: Only generates non-hidden pages now (taking starttime and endttime into account) Thanks to Morten Kjems
	* Sitemap: Does not include pages that have the option "no_search" in the page properties checked
	* Added TypoScript Constant: Append the sitetitle defined in the sys_template ("on" by default!)
	* TypoScript: All wrap constants can now be filled with getData fields
	* TypoScript: All values are HSCed now
	* BE Module: Removed code dependencies on realurl (thanks to Masi)

2008-06-30  Benjamin Mack  <benni@typo3.org>

	* Release of seo_basics 0.5.6
	* Sitemap: Fixed problem with absolute external URLs without prefix
	* Sitemap: Fixed problem with baseURLs, domain records and absRefPrefix, all possible now
	* TCA order: Now compatible with 4.2

2008-02-13  Benjamin Mack  <benni@typo3.org>

	* Release of seo_basics 0.5.5
	* Sitemap: Fixed bug with missing ending brace
	* Release of seo_basics 0.5.4
	* Configuration: Fixed problem showing default titletags for multilanguage sites (Thanks to Andreas Kostner)
	* Sitemap: Added redirect so only main /sitemap.xml URLs are valid now, others are redirected to this one, must be enabled
	* Sitemap: Fixed small issue when pagetype of root was shortcut and shortcut_mode was selected (Thanks to Roman Buechler)
	* Source Formatter: Only active with pagetype 0

2008-01-09  Benjamin Mack  <benni@typo3.org>

	* Release of seo_basics 0.5.3
	* TCA: Added TCA array for tx_seo_titletag field in the pages overlay
	* Manual: Added help for coolURI (thanks to Jan Bednarik and CryAngel)
	* BE: When editing pagetype "shortcut", description & keywords are shown too now
	* BE module: Fixed error when in multilang mode
	* BE module: Fixed coloring in keywords

2007-12-31  Benjamin Mack  <benni@typo3.org>

	* BE module: Syntax highlighting is working again
	* Meta-Tags: Now it also works if only constants are defined
	* Meta-Tags: Constants are noTrimWraps now

2007-12-25  Benjamin Mack  <benni@typo3.org>

	* BE module: Fixed multilanguage problems
	* BE module: Options are now saved in session
	* HTML source formatter: Fixed problems with line breaks and indention (no indention allowed)

ext_emconf.php
<?php

########################################################################
# Extension Manager/Repository config file for ext "seo_basics".
#
# Auto generated 05-07-2011 20:28
#
# Manual updates:
# Only the data in the array - everything else is removed by next
# writing. "version" and "dependencies" must not be touched!
########################################################################

$EM_CONF[$_EXTKEY] = array(
	'title' => 'Basic SEO Features',
	'description' => 'Introduces a separate field for the title-tag per page, easy and SEO-friendly keywords and description editing in a new module as well as a Google Sitemap (XML) and a clean output in the HTML Source code.',
	'category' => 'be',
	'shy' => 0,
	'version' => '0.8.6',
	'dependencies' => '',
	'conflicts' => '',
	'priority' => '',
	'loadOrder' => '',
	'module' => '',
	'state' => 'beta',
	'uploadfolder' => 0,
	'createDirs' => '',
	'modify_tables' => 'pages,pages_language_overlay',
	'clearcacheonload' => 0,
	'lockType' => '',
	'author' => 'Benjamin Mack',
	'author_email' => 'benni@typo3.org',
	'author_company' => '',
	'CGLcompliance' => '',
	'CGLcompliance_note' => '',
	'constraints' => array(
		'depends' => array(
			'realurl' => '0.0.0-0.0.0',
			'typo3' => '4.5.0-6.2.99',
		),
		'conflicts' => array(
		),
		'suggests' => array(
		),
	),
	'_md5_values_when_last_written' => 'a:17:{s:9:"ChangeLog";s:4:"4552";s:22:"class.tx_seobasics.php";s:4:"765c";s:30:"class.tx_seobasics_sitemap.php";s:4:"1ce3";s:16:"ext_autoload.php";s:4:"d1ba";s:21:"ext_conf_template.txt";s:4:"238d";s:12:"ext_icon.gif";s:4:"0858";s:17:"ext_localconf.php";s:4:"db8c";s:14:"ext_tables.php";s:4:"c52f";s:14:"ext_tables.sql";s:4:"e0ac";s:33:"Resources/Private/Language/db.xml";s:4:"0eda";s:14:"doc/manual.sxw";s:4:"fa8b";s:40:"modfunc1/class.tx_seobasics_modfunc1.php";s:4:"d035";s:22:"modfunc1/locallang.php";s:4:"2514";s:29:"modfunc1/js/mootools.v1.11.js";s:4:"4129";s:24:"modfunc1/js/seobasics.js";s:4:"1986";s:20:"static/constants.txt";s:4:"f21e";s:16:"static/setup.txt";s:4:"9902";}',
	'suggests' => array(
	),
);

?>
